### DISCUSSION TITLE
Thread vs BLE Mesh

### DISCUSSION THREAD

**Discussion by wassfila** [HomeSmartMesh/sdk-hsm-sensortag#7] [Q&A] [answered]
@wassfila 

I was thinking, imagine 280  end-devices transmitting data to a server by RADIO

Looks that use THREAD on end-devices to transmit using UDP is faster that the server making a BLE SERVER scanning of end-devices making broadcasting...

Right ?

_Originally posted by @tcpipchip in https://github.com/HomeSmartMesh/sdk-hsm-sensortag/discussions/5#discussioncomment-2350178_

**Comment by wassfila** [comment, marked as answer]
I would say that it is not true. BLE has added a mesh concept to catch up with Thread, and there is no scanning on that mode of end devices one by one, it is similar to listening for broadcasts.
You can search for "Thread vs BLE" to know more about it. My recommendation at the end is the following :

if your devices need to be connected to the user smartphone with direct connection without network infrastructure, then BLE
if devices are permanent monitoring sensors connected to a server, then Thread has a more solid networking due to transparent bridging of network.
I tried to check BLE mesh with IPv6 to compare both, it was much harder at least me, I could not manage, maybe it got easier by now.

Here is a serious and professional benchmarking of 3 technologies Thread, Zigbee and Bluetooth Mesh

https://www.silabs.com/wireless/multiprotocol/mesh-performance

**Comment by tcpipchip** [comment]
BLE make me crazy

Threads much more fast!

**Reply by wassfila** [reply]
You have to be careful of integration differences, according to the silabs study, the difference is not huge.

**Comment by tcpipchip** [comment]
![image](https://user-images.githubusercontent.com/11488460/158243792-e16f9b8f-2e89-46c7-9adf-15394d4322c3.png)

**Reply by wassfila** [reply]
Is that an nRF52 udp thread listener ?

**Comment by tcpipchip** [comment]
![image](https://user-images.githubusercontent.com/11488460/158410632-cf060588-79b8-48d0-add1-8497d683891b.png)
3 modules sending udp packages :)

**Reply by wassfila** [reply]
With 115200 bps serial port, you could easily have a bottleneck on printing. Using native usb cdc could get faster, but yeah throughput is high.

**Comment by tcpipchip** [comment]
Hi, all listeners are the NCP on raspberry, yeah, 115200 is a problem
Next step is to implement a node-red application to listen UDP.

**Comment by tcpipchip** [comment]
My idea is to use 210 threads modules to transmit package each 1 second + some ms, to not allow cillisions

**Comment by tcpipchip** [comment]
Hi, making more tests, now node-red listening 3 UDP packages, no CRASH

See at end of blog 
https://nina-b3-openthread.blogspot.com/2022/03/nina-b302-transmitindo-dados-via-udp.html

**Comment by wassfila** [comment]
@tcpipchip 
congratulation for the great success.
Please keep me updated (but on other discussions tickets) and especially if you need further help with code that I've written, but it feels like that you're the pro now and can even help others ;)
I will lock this discussion because it is deviating from the the topic Thread vs BLE Mesh on which I would not have much to say than the initial provided answer anyway.
